### ISSUE TITLE
Serdes Does Not Work for Items in Object in Array

### ISSUE THREAD

**Issue by YaakovR** [cdimascio/express-openapi-validator#699]
**Describe the bug**
Serdes performs well on properties in an object. However, when the owning object is part of an array, then `serialize` is never called.

**To Reproduce**
1. `git clone https://github.com/cdimascio/express-openapi-validator.git`
2. `npm run deps && npm i`
2. `cd express-openapi-validator/examples/7-response-date-serialization`
3. `npm start`
4. `GET http://localhost:3000/v1/date-time`

The above works fine.

Let's now modify the route in `app.js` to return an array of objects:
```
app.get('/v1/date-time', function (req, res, next) {
  res.json(
    // {
    //   id: 1,
    //   created_at: new Date(),
    // }
    [
      {
        id: 1,
        created_at: new Date(),
      }
    ]
  );
});
```
We'll also modify the schema in `api.yaml`:
```
/date-time:
  get:
    responses:
      200:
        description: date-time handler
        content:
          application/json:
            schema:
#              type: object
#              properties:
#                created_at:
#                  type: string
#                  format: date-time
#                id:
#                  type: number
              type: array
              items:
                type: object
                properties:
                  created_at:
                    type: string
                    format: date-time
                  id:
                    type: number
```

**Actual behavior** 
```
{
    "message": ".response[0].created_at should be string",
    "errors": [
        {
            "path": ".response[0].created_at",
            "message": "should be string",
            "errorCode": "type.openapi.validation"
        }
    ]
}
```

**Expected behavior**
```
[
    {
        "id": 1,
        "created_at": "2022-01-20T17:40:41.816Z"
    }
]
```

**Comment by Jackman3005** [comment]
Did this PR ever make it through? I think I'm having that issue myself now.

**Comment by robertjustjones** [comment]
@Jackman3005 No it did not.  

@cdimascio Can we get #704 reviewed and merged now?

@Fabiencdp

**Comment by srgg** [comment]
I've spent several days digging to the root cause by facing the same issue and happily found that there is a PR already, thank you @robertjustjones 

Gentlemen, when do you think this can be fixed?

**Comment by Fabiencdp** [comment]
We waiting for the approve of the repository owner

**Comment by t0lik** [comment]
@cdimascio Are there any hints about when a release with this fix will be rolled out?

**Comment by cdimascio** [comment]
This is now available in v5.0.0
